tools/harness-thci/OpenThread_BR.py

```diff
@@ -665,7 +665,10 @@ def parse_cache(cache):
             for ba in border_agents:
                 for record in cache[ba.server_name]:
                     if isinstance(record, DNSAddress):
-                        addr = ipaddress.ip_address(record.address)
+                        try:
+                            addr = ipaddress.IPv6Address(record.address)
+                        except Exception:
+                            continue
                         if addr.is_link_local:
```

Comment:
```suggestion
                        if isinstance(addr, ipaddress.IPv6Address) and addr.is_link_local:
```
nit: avoid catching

Reply:
I believe this doesn't work since the instance of `addr` will be of the `ipaddress.ip_address` class.

Reply:
Okay, it actually works. Think to remember I tried to implement it that way at the beginning but somehow removed the check because didn't work for me then. Anyway, will update the PR.

Reply:
```
>>> ipaddress.ip_address('169.254.0.0')
IPv4Address('169.254.0.0')
>>> ipaddress.ip_address('fe80::abcd')
IPv6Address('fe80::abcd')
```